zerver/tests/test_invite.py

```diff
@@ -2784,6 +2915,61 @@ def test_multiuse_invite_without_permission_to_subscribe_others(self) -> None:
         )
         self.assert_json_success(result)
 
+    def test_multiuser_link_with_specified_user_groups_when_cannot_edit_user_groups(self) -> None:
+        hamlet = self.example_user("hamlet")
+        realm = hamlet.realm
+        user_group = create_user_group_in_database("test", [hamlet], realm, acting_user=None)
+        # Only admins are allowed to create and manage user groups.
+        do_set_realm_property(
+            realm,
+            "user_group_edit_policy",
+            Realm.POLICY_ADMINS_ONLY,
+            acting_user=None,
+        )
+
+        members_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.MEMBERS, realm=realm, is_system_group=True
+        )
+        do_change_realm_permission_group_setting(
+            realm, "create_multiuse_invite_group", members_system_group, acting_user=None
+        )
+        self.login("hamlet")
+        result = self.client_post(
+            "/json/invites/multiuse",
+            {
+                "user_group_ids": orjson.dumps([user_group.id]).decode(),
+                "invite_expires_in_minutes": 2 * 24 * 60,
+            },
+        )
+        self.assert_json_error(result, "Insufficient permission")
```

Comment:
I suspect that this test isn't testing what you think it is -- that exception is likely Hamlet not having permission to make multi-use invites at all.

Reply:
(Generally good test design would be to first verify that making an invite works without the `user_group_ids` parameter, and then add it, so that the test construction makes it clear which variable is the thing causing the failure)

Reply:
Done 👍